.github/workflows/ci.yml

```diff
@@ -4,13 +4,19 @@ on: [push, pull_request]
 jobs:
   run:
     runs-on: ubuntu-latest
+    outputs:
+      artifact: ${{ steps.pack.outputs.tarball }}
     steps:
       - uses: actions/checkout@v4
       - run: corepack enable
       - run: pnpm install
-      - run: pnpm run build
```

Comment:
I don't love that this is being removed - the purpose of this job is validate the project is set up right, and the build step is part of that. Sure, it's now part of pnpm pack, which is a good new thing, but it's not very visible.

Reply:
I can separate them

Reply:
Just realized by re-adding `npm run build` technically we run the `build` script twice.

Reply:
I'm ok with running it twice, I think it's better than removing the build step from here.

I just pushed what I think is a compromise solution between this and #111 - closer to yours, which I think is better since it avoids a dependency on some external service. But also I didn't love how this one has to use upload-artifact and download-artifact.

Now, it leaves this job untouched, and just checks out, installs, and packs in the are-the-types-wrong job.

Tradeoffs:

- doesn't impact the main `ci` job 👍 
- doesn't upload-artifact and download-artifact 👍 
- doesn't depend on pkg.pr.new 👍 
- does a full `pnpm install` so has my made-up node_modules risk 👎 
- builds twice 👎 
- runs in parallel and GitHub is paying for it to build twice anyway 👍👍👍 
- my node_modules risk is almost certainly bogus though 👍

Reply:
Works for me! Does it need anything else?